ccf_transaction_processor/Readme.md

```diff
@@ -3,40 +3,172 @@ Licensed under Creative Commons Attribution 4.0 International License
+before executing the above command:
+
+A. In the setup_nosgx.sh file, modify the command `sudo add-apt-repository ppa:ansible/ansible -y` by adding the `-E` option. The modified command looks like `sudo -E add-apt-repository ppa:ansible/ansible -y`. Add a `sudo` prefix to the `ansible-playbook ccf-dependencies-no-driver.yml` command appearing in the same file.
+
+B. In ccf-dependencies-no-driver.yml (present under the same folder), add the environment option for proxies
+(see https://docs.ansible.com/ansible/latest/user_guide/playbooks_environment.html for reference)
+
+```bash
+environment:
+    http_proxy: <specify-http-proxy-here>
+    https_proxy: <specify-https-proxy-here>
+```
+
+C. Add the repo used by ansible scripts for installing python 3.7
+
+```bash
+sudo add-apt-repository ppa:deadsnakes/ppa
+```
+
+## Build CCF & PDO-TP
+
+CCF uses a combination of cmake& ninja to build the applcaition. Execute the following steps to complete the build process:
+
+A. Add the following lines to the CMakelists.txt found at $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF.
+    (look for other add_ccf_app in this file, add just above or below this)
+
+```bash
+add_ccf_app(
+pdoenc SRCS ../transaction_processor/pdo_tp.cpp
+../transaction_processor/verify_signatures.cpp
+)
+```
+
+B. Create the build folder
+
+```bash
+mkdir $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF/build
+```
+
+C. Set the build flags
+
+```bash
+cd $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF/build
+cmake -GNinja -DCOMPILE_TARGETS=virtual -DBUILD_END_TO_END_TESTS=OFF -DBUILD_SMALLBANK=OFF \
+                -DBUILD_TESTS=OFF-DBUILD_UNIT_TESTS=OFF ..
+```
+
+D. Build using ninja
+
+```bash
+ninja
+```
+
+If the VM has a small memory (< 4GB), it might be required to reduce the parallelism in the build process using the `-j` flag for the `ninja` command, say ninja -j2 or even ninja -j1 (the default seems to be ninja -j4).
+
+If build is successful, the libpdoenc.virtual.so library is created inside the build folder (ignore any other targets that get created)
+
+## Deploy CCF with PDO-TP as the application
+
+The following commands start a single node CCF under the virtual enclave mode.  PDO-TP is the hosted application.
+
+```bash
+cd $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF/build
+python3.7 -m venv env
+source env/bin/activate
+pip install -q -U -r ../tests/requirements.txt
+python ../tests/start_network.py --gov-script ../src/runtime_config/gov.lua  --label pdo_tp -e virtual --package libpdoenc.virtual.so --node <ip-address:6006>
+```
+
+A single node CCF instance hosting the PDO TP will be available for business use @ ip-address:6006. Here `ip-address` is the address of this VM that can be used to ping this machine from other VMs (this is our use case). Set ip-address to 127.0.0.1 for local testing. If you are behind a proxy, add ip-address to the list of no_proxy bash env variable.
+
+## Share CCF Keys
```

Comment:
At first, I thought this section was going to talk about the ledger verifying key. Maybe we should clarify that this section talks about the TLS keys:

```suggestion
## Share CCF TLS Keys
```

Reply:
thank you, added